astropy/table/index.py

```diff
@@ -967,6 +1027,20 @@ def __getitem__(self, item):
         return super().__getitem__(item)
 
 
+@deprecated(
+    since="7.2.0",
+    message="""\
+Calling `Table.loc/iloc/loc_indices[index_id, item]` to select `item` from index
+`index_id` is deprecated. Instead select the index using the syntax
+`Table.loc/iloc/loc_indices.with_index(index_id)[item]`.
+""",
+)
+def interpret_item_as_index_id_and_item(item: tuple) -> tuple:
+    """Interpret the item as a (index_id, item) tuple."""
```

Comment:
I find this confusing. Could you explain the difference between `item` (the input), and `item` (the last element on the output tuple) ?
I can include the explanation later if you don't want to spin CI again here :)

Reply:
This is exactly the confusion that is being deprecated away. You can do `t.loc[index_id, actual_item]` which translates to `t.loc.with_index(index_id)[actual_item`. But before that we have `t.loc.__getitem__(item)`, and that is the input `item` here.

The only reason for this little function is so that it can be deprecated now and then removed in 9.0. Previously this was just a line of code in an `if` block.